src/backend/distributed/operations/worker_shard_copy.c

```diff
@@ -344,21 +345,80 @@ ShardCopyDestReceiverDestroy(DestReceiver *dest)
 }
 
 
+/*
+ *  GenerateColumnListFromTupleDesc function creates and returns a comma seperated column names string  to be used in COPY
+ *  and SELECT statements when copying a table. The COPY and SELECT statements should filter out the GENERATED columns since COPY
+ *  statement fails to handle them. Iterating over the attributes of the table we also need to skip the dropped columns.
+ */
+const char *
+GenerateColumnListFromTupleDesc(TupleDesc tupDesc)
+{
+	StringInfo columnList = makeStringInfo();
+	bool firstInList = true;
+
+	for (int i = 0; i < tupDesc->natts; i++)
+	{
+		Form_pg_attribute att = TupleDescAttr(tupDesc, i);
+		if (att->attgenerated || att->attisdropped)
+		{
+			continue;
+		}
+		if (!firstInList)
+		{
+			appendStringInfo(columnList, ",");
+		}
+
+		firstInList = false;
+
+		appendStringInfo(columnList, "%s", quote_identifier(NameStr(att->attname)));
+	}
+
+	return columnList->data;
+}
+
+
+/*
+ *  GenerateColumnListFromRelationName function is a wrapper for GenerateColumnListFromTupleDesc.
+ */
+const char *
+GenerateColumnListFromRelationName(const char *relationName, const char *schemaName)
```

Comment:
Previous name is not very descriptive and generally schemaname comes before tablename.
```suggestion
CopyableColumnNamesFromRelationName(const char *schemaName, const char *relationName)
```